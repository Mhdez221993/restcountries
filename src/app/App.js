import React from 'react';
import { BrowserRouter as Router, Route, Routes } from 'react-router-dom';
import NavBar from '../features/nav/Nav';
import './App.css';
import routes from './routes';

function App() {
  return (
    <Router basename={process.env.PUBLIC_URL}>
      <div className="app">
        <NavBar routes={routes} />
        <Routes>
          {routes.map(
            ({ path, component, name }) => (
              <Route exact path={path} element={component} key={name} />
            ),
          )}
        </Routes>
      </div>
    </Router>
  );
}

export default App;
