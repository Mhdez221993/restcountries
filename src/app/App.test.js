import { shallow } from 'enzyme';
import React from 'react';
import App from './App';
describe('App', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<App debug />);

    expect(component).toMatchSnapshot();
  });

  it('should render banner text correctly with given strings', () => {
    const routes = ['one', 'two'];
    const component = shallow(<App routes={routes} />);
    expect(component).toMatchSnapshot();
  });
});
