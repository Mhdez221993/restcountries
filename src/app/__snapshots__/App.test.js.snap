// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render banner text correctly with given strings 1`] = `ShallowWrapper {}`;

exports[`App should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;

exports[`App should render correctly with no props 1`] = `ShallowWrapper {}`;

exports[`MyComponent should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;
